apps/client/src/components/molecules/timer/Timer.component.tsx

```diff
@@ -0,0 +1,137 @@
+import { ReactElement, useEffect, useRef, useState } from 'react';
+import { TextSize, TextType } from 'apps/client/src/models';
+import cs from 'classnames';
+
+import { Text } from '../../atoms';
+
+export interface TimerProps {
+  time: number; // in seconds
+  onTimeEnd: () => void;
+}
+
+/**
+ *
+ * @param time - Time in seconds
+ * @param [onTimeEnd] - Callback function to be called when time ends (optional)
+ */
+export function Timer({
+  time,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  onTimeEnd = (): void => {},
+}: TimerProps): ReactElement {
+  const startTimestamp = useRef(Date.now());
+  const previousTimestamp = useRef(Date.now());
+  const [seconds, setSeconds] = useState(time);
+  const [milliseconds, setMilliseconds] = useState(0);
+  const circleIndicator = useRef<SVGCircleElement>(null);
+  const timerEnded = useRef(false);
+
+  useEffect(() => {
+    const updateTimer = (): void => {
+      if (timerEnded.current) {
+        return;
+      }
+
+      const now = Date.now();
+
+      // Optimizing for high refresh rate screens
+      if (now - previousTimestamp.current < 10) {
```

Comment:
What's done here ? can you put up a more detailed comment

Reply:
Fixed in #114 under eb9f3ff9efcd4c0a6cff8a95f942f0fcc6032ab8